projects/aca-content/src/lib/components/search/search-input-control/search-input-control.component.spec.ts

```diff
@@ -30,59 +30,60 @@ import { NO_ERRORS_SCHEMA } from '@angular/core';
 describe('SearchInputControlComponent', () => {
   let fixture: ComponentFixture<SearchInputControlComponent>;
   let component: SearchInputControlComponent;
-
   beforeEach(() => {
     TestBed.configureTestingModule({
       imports: [AppTestingModule, SearchInputControlComponent],
       schemas: [NO_ERRORS_SCHEMA]
     });
-
     fixture = TestBed.createComponent(SearchInputControlComponent);
     component = fixture.componentInstance;
     fixture.detectChanges();
   });
 
-  it('should emit submit event on searchSubmit', () => {
-    component.searchTerm = 'mock-search-term';
+  it('should emit submit event if form is valid', () => {
+    component.searchTerm = 'valid';
+    spyOn(component.submit, 'emit');
+
+    component.searchSubmit();
+
+    expect(component.submit.emit).toHaveBeenCalledWith('valid');
+  });
 
-    let submittedSearchTerm = '';
-    component.submit.subscribe((searchTerm) => (submittedSearchTerm = searchTerm));
+  it('should not emit submit event if form is invalid', () => {
+    component.searchTerm = '';
+    spyOn(component.submit, 'emit');
 
     component.searchSubmit();
-    expect(submittedSearchTerm).toBe('mock-search-term');
+
+    expect(component.submit.emit).not.toHaveBeenCalled();
   });
 
   it('should emit searchChange event on inputChange', () => {
     let emittedSearchTerm = '';
     component.searchChange.subscribe((searchTerm) => (emittedSearchTerm = searchTerm));
     component.searchTerm = 'mock-search-term';
-
     expect(emittedSearchTerm).toBe('mock-search-term');
   });
 
   it('should emit searchChange event on clear', () => {
     let emittedSearchTerm: string = null;
     component.searchChange.subscribe((searchTerm) => (emittedSearchTerm = searchTerm));
-
     component.clear();
     expect(emittedSearchTerm).toBe('');
   });
 
   it('should clear searchTerm', () => {
     component.searchTerm = 'c';
     fixture.detectChanges();
-
     component.clear();
     expect(component.searchTerm).toBe('');
   });
 
   it('should check if searchTerm has a length less than 2', () => {
```

Comment:
This unit test fails now: 
<img width="1712" height="691" alt="image" src="https://github.com/user-attachments/assets/2631a3a0-9ae7-449b-aa5f-ef9b04564e51" />